File: LeetCode Solutions/19. Remove Nth Node From End of List/solution19.py
# Definition for singly-linked list.
# class ListNode:
# def __init__(self, val=0, next=None):
# self.val = val
# self.next = next

class Solution:
	def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
	root = ListNode(-1, head); tail = root
	while n > 1:
		head = head.next
		n-=1
	while head.next:
		head = head.next
		tail = tail.next
	tail.next = tail.next.next
	return root.next

File: LeetCode Solutions/19. Remove Nth Node From End of List/solution19.java
/**
* Definition for singly-linked list.
* public class ListNode {
* int val;
* ListNode next;
* ListNode() {}
* ListNode(int val) { this.val = val; }
* ListNode(int val, ListNode next) { this.val = val; this.next = next; }
* }
*/



class Solution {
	public ListNode removeNthFromEnd(ListNode head, int n) {
		ListNode root = new ListNode(-1, head), tail = root;
		while (n > 1) {
			head = head.next;
			n--;
		}
	
		while (head.next != null) {
			head = head.next;
			tail = tail.next;
		}
	
		tail.next = tail.next.next;
		return root.next;
	}
}